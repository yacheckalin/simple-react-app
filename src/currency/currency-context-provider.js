import React, { useContext, useState } from "react";

export const CurrencyContext = React.createContext();
export const useCurrencyContext = () => useContext(CurrencyContext);

let currencyData = [
  {
    favorite: false,
    asset: "EUR/USD",
    quote: "1.12",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "USD/JPY",
    quote: "107.95",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "GBP/USD",
    quote: "1.238",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "AUD/USD",
    quote: "0.7048",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "USD/CHF",
    quote: "0.9888",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "USD/CAD",
    quote: "1.31",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
];

const CurrencyContextProvider = ({ children }) => {
  const [data, setData] = useState(currencyData);

  const handleFavoriteChange = (index) => {
    if (currencyData[index]) {
      currencyData[index].favorite = !currencyData[index].favorite;
      setData([...currencyData]);
    }
  };
  return (
    <CurrencyContext.Provider
      value={{ data, setFavorite: handleFavoriteChange }}
    >
      {children}
    </CurrencyContext.Provider>
  );
};

export default CurrencyContextProvider;
