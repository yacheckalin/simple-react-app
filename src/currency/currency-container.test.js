import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import CurrencyContainer from "./currency-container";
import * as CurrencyContext from "./currency-context-provider";

describe("<CurrencyContainer />", () => {
  const mockedData = [
    {
      favorite: false,
      asset: "EUR/USD",
      quote: "1.12",
      date: new Intl.DateTimeFormat("en-US", {
        year: "numeric",
        month: "long",
        day: "2-digit",
      }).format(Date.now()),
    },
    {
      favorite: false,
      asset: "USD/JPY",
      quote: "107.95",
      date: new Intl.DateTimeFormat("en-US", {
        year: "numeric",
        month: "long",
        day: "2-digit",
      }).format(Date.now()),
    },
    {
      favorite: false,
      asset: "GBP/USD",
      quote: "1.238",
      date: new Intl.DateTimeFormat("en-US", {
        year: "numeric",
        month: "long",
        day: "2-digit",
      }).format(Date.now()),
    },
    {
      favorite: false,
      asset: "AUD/USD",
      quote: "0.7048",
      date: new Intl.DateTimeFormat("en-US", {
        year: "numeric",
        month: "long",
        day: "2-digit",
      }).format(Date.now()),
    },
    {
      favorite: false,
      asset: "USD/CHF",
      quote: "0.9888",
      date: new Intl.DateTimeFormat("en-US", {
        year: "numeric",
        month: "long",
        day: "2-digit",
      }).format(Date.now()),
    },
    {
      favorite: false,
      asset: "USD/CAD",
      quote: "1.31",
      date: new Intl.DateTimeFormat("en-US", {
        year: "numeric",
        month: "long",
        day: "2-digit",
      }).format(Date.now()),
    },
  ];

  test("displays and renders properly", () => {
    const contextValues = { data: mockedData, setFavorite: jest.fn() };
    jest
      .spyOn(CurrencyContext, "useCurrencyContext")
      .mockImplementation(() => contextValues);
    const wrapper = shallow(<CurrencyContainer />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
