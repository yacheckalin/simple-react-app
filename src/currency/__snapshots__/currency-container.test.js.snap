// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyContainer /> displays and renders properly 1`] = `
<styled.table>
  <thead>
    <tr>
      <th>
         
      </th>
      <th>
        currency.title
      </th>
      <th>
        currency.quotation
      </th>
      <th>
        currency.date
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
      onClick={[Function]}
    >
      <td>
        <BsStar />
      </td>
      <td>
        EUR/USD
      </td>
      <td>
        1.12
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="1"
      onClick={[Function]}
    >
      <td>
        <BsStar />
      </td>
      <td>
        USD/JPY
      </td>
      <td>
        107.95
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="2"
      onClick={[Function]}
    >
      <td>
        <BsStar />
      </td>
      <td>
        GBP/USD
      </td>
      <td>
        1.238
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="3"
      onClick={[Function]}
    >
      <td>
        <BsStar />
      </td>
      <td>
        AUD/USD
      </td>
      <td>
        0.7048
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="4"
      onClick={[Function]}
    >
      <td>
        <BsStar />
      </td>
      <td>
        USD/CHF
      </td>
      <td>
        0.9888
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
         
      </td>
    </tr>
    <tr
      key="5"
      onClick={[Function]}
    >
      <td>
        <BsStar />
      </td>
      <td>
        USD/CAD
      </td>
      <td>
        1.31
      </td>
      <td>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="YYYY-MM-DD"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </td>
      <td>
         
      </td>
    </tr>
  </tbody>
</styled.table>
`;
